### PR TITLE
Added support for the Waveshare ESP32-S3-Zero and GRB Status LED

### PR DESCRIPTION
**Pull request by Axeia**
Please note I'm just a hobbyist and do not work for Waveshare.

I believe this contains all that's needed to add support for the Waveshare ESP32-S3-Zero board to circuitpython.
I have successfully created a build and flashed it onto my device and have verified all pins are correctly named and working.

### CODE REVIEW

**Comment by deshipu** [comment]
I compiled it and flashed, and it seems to be working.

The neopixel uses GRB colors instead of RGB, so red and green are swapped, but there doesn't currently seem to be an option to handle that.

**Comment by deshipu** [comment]
We might want to enable the espcamera module back, there is no reason why it wouldn't work with this board, there are enough pins, and external ram.

**Review by deshipu — DISMISSED** [review]

File: ports/espressif/boards/waveshare_esp32_s3_zero/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,20 @@
+USB_VID = 0x303a
+USB_PID = 0x1fe8
```
Comment (tannewt) [inline ports/espressif/boards/waveshare_esp32_s3_zero/mpconfigboard.mk:2]: Where does this PID come from? I don't see it listed here: https://github.com/espressif/usb-pids/blob/main/allocated-pids.txt

**Review by tannewt — CHANGES_REQUESTED** [review]
Please get a proper USB PID for this.

File: ports/espressif/boards/waveshare_esp32_s3_zero/mpconfigboard.mk (hunk 1 above)
Comment (Axeia) [reply ports/espressif/boards/waveshare_esp32_s3_zero/mpconfigboard.mk:2]: _We discussed on github but I'll mention it here for the sake of completeness. I got it from running lsusb under Linux. I think this was after I already put circuitpython on it though so very likely not what it should be_

I've contacted Waveshare to see if they want to request a PID from espressif or if I can/should on their behalf. No response so far.

**Comment by tannewt** [comment]
I disabled bleio (via `CIRCUITPY_BLEIO = 0`) on S3 boards with little flash because BLE support on ESP is incomplete.

**Comment by Axeia** [comment]
Pushed the update with my GRB LED edit and updated the USB PID to one given by espressif specifically for this board.

**Comment by Axeia** [comment]
> Thanks for getting a new PID! You'll need to fix the pre-commit test before we can merge. It should be auto-fixable on your machine. See pre-commit info here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code

Ah I see what happened - I got lazy and edited it on github directly so not everything got triggered properly. Updated from my own machine now - thank you for your patience.

**Comment by Axeia** [comment]
Not quite sure what's going wrong now. The pre-commit hook wanted to run some translation with generated additional files and now one of those additional files is throwing errors?

**Comment by dhalbert** [comment]
If you look here: https://github.com/adafruit/circuitpython/pull/8468/files, you'll see that you have different versions (probably outdated) of the `ulab` module and a library module. These should not be in your commits. These should be brought up to date. The only files changed in your commits should be the files you are adding. That's causing the build problems.

I notice that you forked for this PR, but are working on `main`. Instead. you can fork once, and then reuse the fork. But always create a new branch for a PR. The suggest workflow is described here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github.

**Pull request closed without merging**